### PR TITLE
Funnel refactor

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- this is based on insight history
- refactors funnels to not use models but follow filter pattern instead
- adds polling to funnels
- fixes caching. previously the way filters were compared were different so the cache was not always matched with the intended element

*If this affects the front-end, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [ ] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)
- [ ] fix default dashboard funnel filter

### CODE REVIEW

**Comment by EDsCODE** [comment]
#1379 had already been approved and just reverted. this PR is for merging the funnel refactor in the insight history branch so that both features/functionality can be merged into master together. 

**Important consideration**: This pr contains 2 data migrations that alter dashboarditems and repopulate the funnel items as dashboard items. This ran successfully on the replica db @fuziontech provided and it runs well on demo data; however, there are a few conditions I added after testing it on the replica. If some action_ids and funnel_ids weren't queryable (ex: action id=7 doesn't exist in db), I dropped it from the funnel. I wasn't able to fully test the effects of this because running a backend with my machine on the replica db was too slow

**Review by timgl — COMMENTED** [review]
Mostly looks good, few things that confused me

1. the fact I have to "save funnels", it'd be nicer if it just worked immediately like everything else
2. When I click "save funnels", I expected it to get saved into the list below

**Comment by EDsCODE** [comment]
Also, now we'll want to wait on #1430 before merging this! @timgl still good to review and I'll handle the conflicts when #1430 is in master

File: posthog/migrations/0078_update_dashboard_funnel_filters.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                    filters["actions"][index]["name"] = name
+                except:
+                    del filters["actions"][index]
+
+        item.filters = filters
+        item.save()
+
+
+def reverse(apps, schema_editor):
+    pass
```
Comment (EDsCODE) [inline posthog/migrations/0078_update_dashboard_funnel_filters.py:36]: Need to add reversals incase we need to rollback

File: posthog/migrations/0077_move_funnels_to_insights.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            name=item.name,
+            deleted=item.deleted,
+            filters=filters,
+            created_by=item.created_by,
+            saved=True,
+        )
+
+
+def reverse(apps, schema_editor):
+    pass
```
Comment (EDsCODE) [inline posthog/migrations/0077_move_funnels_to_insights.py:38]: here too

**Review by timgl — COMMENTED** [review]
Almost! I think there _should_ be a save button, and it should save it into history. There's an argument to be made that that should happen for every view, but this one is more obvious as the other saved funnels are right there and it's not obvious how to add new ones

**Comment by timgl** [comment]
I'm sure I'm being thick but I still can't see a save button, only a calculate button

**Comment by EDsCODE** [comment]
oops my comment was left in a tab unsent

I was going to suggest we make the funnel button 'calculate' atm and we remove the saved funnels panel (haven't done this yet). In a follow-up PR i can add the saved <insight> panel to every type so it's more consistent. otherwise I think with the history tab and the saved funnels panel people will be really confused why there are so any places that show the same exact thing

**Pull request merged**